conjure/src/test/java/com/palantir/conjure/cli/ConjureCliTest.java

```diff
@@ -185,14 +186,30 @@ public void generatesCleanError_unique_name() {
         assertThat(outputFile).doesNotExist();
     }
 
+    @Test
+    public void generatesCleanError_invalid_json() {
+        String[] args = {"compile", "src/test/resources/invalid-json.yml", outputFile.getAbsolutePath()};
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter printWriter = new PrintWriter(stringWriter);
+        ConjureCli.prepareCommand().setErr(printWriter).execute(args);
+        printWriter.flush();
+        assertThat(stringWriter.toString())
+                .isEqualTo("Error while parsing src/test/resources/invalid-json.yml:\n"
+                        + "Cannot build FieldDefinition, some of required attributes are not set [type]\n"
```

Comment:
This might be a nit, but I feel like I'm missing some sort of separate between the nested causes; canonically, the stack traces use "Caused by:" and a giant tab for the text below; maybe we shouldn't strip the "Caused by", since it's so ingrained in how most people read stacktraces?

Reply:
These intentionally aren't caused by's. Ideally we'd replace this with a proper error tracking system which would allow for proper indentation, but we were abusing java exceptions to communicate parsing issues.

If you feel strongly about this, I could add back a "Caused by:" prefix on every line, but I don't think it'll improve readability (even if the repetition is a little annoying/confusing).

Samples:

```
Error while parsing src/test/resources/invalid-json.yml:
Caused by: Cannot build FieldDefinition, some of required attributes are not set [type]
  @ types -> definitions -> objects -> InvalidJson -> union -> optionA
Caused by: Cannot build FieldDefinition, some of required attributes are not set [type]
```

```
Type, error, and service names must be unique across locally defined and imported types/errors:
Caused by: Found duplicate name: test.api.ConflictingName
Known names:
 - test.api.UniqueName
 - test.api.UniqueName2
 - test.api.ConflictingName
```

```
Encountered error trying to parse file 'src/test/resources/simple-error.yml'
Unknown LocalReferenceType: TypeName{name=UnknownType}
```